.github/workflows/go.yml

```diff
@@ -25,8 +25,10 @@ jobs:
       - name: Build and Test
         run: make
 
-      - name: Upload coverage
-        run: bash <(curl -s https://codecov.io/bash)
+      - uses: codecov/codecov-action@v4
+        with:
+          codecov_yml_path: ./codecov.yml
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
I'm not sure we've set up codecov tokens on other repos in HL previously.

I think this PR is pointing out that codecov is trying to deprecate the ability to use it without a token, and as such we might need one.

That might need a conversation with the HL team potentially, as while I believe the free tier would be fine still on codecov.io, there would need to be a user that could own the github account that is associated with the token.

Reply:
The repo has a secret `CODECOV_TOKEN` set for the Github Action to use and the previous `bash` step used the token already. I have not idea who/how that token ended up in the repository, doesn't seem to be a HL thing based on not seeing anything on discord. I do wonder how we created an account against codecov and generated a token

Reply:
Conversation with HL https://discord.com/channels/905194001349627914/905205991073804308/1252995889522282599